backend/app/auth.py

```diff
@@ -83,24 +83,72 @@ async def _get_user(
     return _get_user
 
 
-async def _verify_token(token: str) -> dict:
+async def is_admin(token: str = Security(oidc)):
+    user_info = await _get_userinfo(token)
+
+    if not has_admin_role(user_info):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="You don't have enough privileges",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+
+
+def has_admin_role(user_info: dict) -> bool:
+    user_client_roles = (
+        user_info.get("resource_access", {})
+        .get(settings.AIOD_KEYCLOAK.CLIENT_ID, {})
+        .get("roles", [])
+    )
+    return "admin_access" in user_client_roles
+
+
+async def _get_userinfo(token: str) -> dict:
+    if token is None:
+        _raise_requires_auth()
+
+    token = token.replace("Bearer ", "")
+
     try:
-        token = token.replace("Bearer ", "")
-        return keycloak_openid.userinfo(token)  # perform a request to keycloak
+        return keycloak_openid.userinfo(token)
     except KeycloakError as e:
-        error_msg = e.error_message
-        error_detail = "Invalid authentication token"
+        _raise_invalid_token(e)
 
-        if isinstance(error_msg, bytes):
-            error_msg = error_msg.decode("utf-8")
 
-        if error_msg:
-            error_detail = f"{error_detail}: '{error_msg}'"
+async def _get_introspect(token: str) -> dict:
```

Comment:
You are right, it is not being used for now. It fetched more detailed info from Keycloak and I initially planned to use that instead of userinfo. At the end of the day I sticked with simple userinfo for now. I wanted to keep it the as we might change that in the near future, but I am fine with removing it for now.